docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+
+The following 2 UML diagrams show the different types of `CliCommand` components used in the application.
+
+This diagram describes the `CliCommand`s related to the Timetable mode.
+
+![UML diagram for Timetable Commands](diagrams/class/jpeg/timetable_commands.jpg)
+
+And this diagram describes the `CliCommand`s related to the Notebook mode.
+
+![UML diagrams for Notebook Commands](diagrams/class/jpeg/notebook_commands.jpg)
+
+// TODO: add the new commands for "Find", "Tag", "SortByDate".
+
+### 3.4. Tasks Component
+
+![UML Diagram from Task Component](diagrams/class/jpeg/taskComponent.jpg)
+
+Figure []. Structure of Tasks Component
+
+The `Tasks` component,
+
+- stores an array of `Task` objects that represents the deadline of the user.
+- together with the `Notebooks` component form the `AppState` component.   
+
+The `TaskList` class,
+
+- has methods to add a new `Task` object and remove existing one at an index.
+
+### 3.5. Notebooks Component
+
+![UML diagram for Notebooks Component](diagrams/class/jpeg/notebooks_simplified.jpg)
+
+Figure []. Structure of Notebooks Component
+
+The `Notebooks` class,
+
+- stores a `NotebookShelf` object that contains a list of `Notebook` objects.
+- together with the `Tasks` component form the `AppState` component.
+
+The `NotebookShelf` class,
+
+- has methods to add new `Notebook` object and remove existing one.
+- has methods to search through to find a `Notebook` object with matching titles.
+
+The `Notebook` class,
+
+- contains a title and a list of `Section` objects.
+- has methods to add new `Section` object or remove existing one.
+- has a `tag` field that user can be set and get.
+
+The `Section` object,
+
+- contains a title and list of `Page` objects.
+- has methods to add new `Page` object or remove existing one.
+- has a `tag` field that user can be set and get.
+
+The `Page` object,
+
+- contains a title, and the content of the page as a String object.
+- has a `tag` field that user can be set and get.
+
+### 3.6. Storage Component
+
+![UML diagram for Storage](diagrams/class/jpeg/Storage_UML_class.jpg)
+
+The `Storage` component,
+
+* Contains the method `saveToFile` to save the current AppState of the application in the `notebooks.txt` and `tasks.txt` files.  
+
+* Contains the method `readFromFile` to load up data containing the input of the user's previous session on Zer0Note.  
+
+* Saves the name of the user in a `nameOfUser.txt` and reads it back in the form of personalised messages.   
+
+## 4. Implementation
+
+The following section describes the implementation of certain key features in the current version of Zer0Note. It also
+provides some background into our (the original developers of Zer0Note) thinking and the rationale behind the
+decisions.
+
+### 4.1. Mode Switch Feature
+
+#### 4.1.1. Implementation
+
+The mode switch mechanism is facilitated by `AppState`. It contains an `AppMode` object and can be accessed from
```

Comment:
I believe it might be better if you had talked about the different modes before this feature as it is a bit difficult to keep track of the different modes that are present in the app?